utam-compiler/src/main/java/utam/compiler/translator/JsonCompilerConfig.java

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root
+ * or https://opensource.org/licenses/MIT
+ */
+package utam.compiler.translator;
+
+import static utam.core.declarative.translator.UnitTestRunner.NONE;
+import static utam.core.declarative.translator.UnitTestRunner.validateUnitTestDirectory;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import utam.compiler.UtamCompilationError;
+import utam.compiler.translator.DefaultSourceConfiguration.RecursiveScanner;
+import utam.compiler.translator.DefaultSourceConfiguration.ScannerConfig;
+import utam.core.declarative.translator.ProfileConfiguration;
+import utam.core.declarative.translator.TranslatorSourceConfig;
+import utam.core.declarative.translator.TranslatorTargetConfig;
+import utam.core.declarative.translator.UnitTestRunner;
+
+/**
+ * JSON based config for UTAM compiler, class is public because of inner classes that can be used in
+ * downstream projects
+ *
+ * @author elizaveta.ivanova
+ * @since 234
+ */
+public class JsonCompilerConfig {
```

Comment:
this is not new file, renamed JsonBasedCompilerConfigTests.java

Reply:
Understood. Approved for this file.